contracts/cw-proposal-single/src/proposal.rs

```diff
@@ -118,6 +129,18 @@ impl Proposal {
         if self.allow_revoting && !self.expiration.is_expired(block) {
             return false;
         }
+        // If the min voting period is set and not expired the
+        // proposal can not yet be passed. This gives DAO members some
+        // time to remove liquidity / scheme on a recovery plan if a
+        // single actor accumulates enough tokens to unilaterally pass
+        // proposals.
+        if !self
+            .min_voting_period
+            .map(|min| min.is_expired(block))
+            .unwrap_or(true)
+        {
+            return false;
+        }
```

Comment:
Not sure if it is because I'm new to Rust and not used to the idioms of testing conditions wrapped into `Option`, but I found this somewhat difficult to read. IMHO, here's an alternative that looks more explicit/easier to understand:

```rust
if let Some(min) = self.min_voting_period {
    if !min.is_expired(block) {
        return false;
    }
}
```

Reply:
Great point. :) Was being a little fancy here. Will update. Your suggestion is much more clean.